src/coreclr/vm/object.cpp

```diff
@@ -1545,7 +1521,23 @@ void StackTraceArray::CopyFrom(StackTraceArray const & src)
     }
     CONTRACTL_END;
 
-    m_array = (I1ARRAYREF) AllocatePrimitiveArray(ELEMENT_TYPE_I1, static_cast<DWORD>(src.Capacity()));
+    m_array = (I1ARRAYREF) AllocatePrimitiveArray(ELEMENT_TYPE_I1, static_cast<DWORD>(src.Get()->GetNumComponents()));
```

Comment:
We may want to only allocate enough space for the entries that are filled in. We should not need to clone the empty slots.

Reply:
Or even better - if you take my other suggestion, we can freeze the stack by setting its thread to `(Thread*)1`. It would make the storing and restoring of ExceptionDispatchInfo allocation-free. The copy would be created lazily only once we need to append more frames to the stack trace after the restored ExceptionDispatchInfo is rethrown.